src/linux/system.rs

```diff
@@ -420,25 +417,16 @@ fn copy_from_file(entry: &Path) -> Vec<String> {
 fn get_all_disks() -> Vec<Disk> {
     #[allow(or_fun_call)]
     let content = get_all_data("/proc/mounts").unwrap_or(String::new());
-    let disks: Vec<_> = content.lines()
-                               .filter(|line| line.trim_left()
-                                                  .starts_with("/dev/sd"))
-                               .collect();
-    let mut ret = Vec::with_capacity(disks.len());
-
-    if disks.len() == 1 {
-        let info: Vec<_> = disks[0].split(' ').collect();
-        if info.len() < 3 {
-            return ret;
-        }
-        ret.push(disk::new_disk("sda", info[1], info[2]));
-    } else {
-        for line in disks {
-            let info: Vec<_> = line.split(' ').collect();
-            if info.len() < 3 {
-                continue
-            }
-            ret.push(disk::new_disk(&info[0][5..], info[1], info[2]));
+    let disks =content.lines()
+        .filter(|line| line.trim_left()
+            .starts_with("/dev/sd"));
```

Comment:
Are you sure? The .starts_with is nested within the filter call, not chaining after it.

The spacing around = looks wrong though.